applications/FluidDynamicsApplication/custom_elements/symbolic_stokes.cpp

```diff
@@ -198,6 +198,85 @@ void SymbolicStokes<TElementData>::AddBoundaryTraction(
     noalias(rRHS) += prod(rData.lhs,values);
 }
 
+template <class TElementData>
+void SymbolicStokes<TElementData>::Calculate(
+    const Variable<double> &rVariable,
+    double &rOutput,
+    const ProcessInfo &rCurrentProcessInfo)
+{
+    KRATOS_TRY
+    rOutput = 0.0;
+
+    if (rVariable == HEAT_FLUX) {
+        TElementData data;
+        data.Initialize(*this, rCurrentProcessInfo);
+        rOutput = 0.0;
+        // Shape functions
+        const GeometryType &r_geometry = this->GetGeometry();
+        const unsigned int num_nodes = r_geometry.PointsNumber();
+        Vector data_N(num_nodes);
+        for (unsigned int i = 0; i < num_nodes; i++) {
+            data_N[i] = 1.0 / (double)num_nodes;
+        }
+        data.N = data_N;
+        // Shape functions gradients
+        const GeometryData::IntegrationMethod integration_method = GeometryData::GI_GAUSS_2;
+        ShapeFunctionDerivativesArrayType shape_derivatives;
+        Vector DetJ;
+        r_geometry.ShapeFunctionsIntegrationPointsGradients(shape_derivatives, DetJ, integration_method);
+        data.DN_DX = shape_derivatives[0]; // Note: Valid only for linear elements
+        // Compute strain and stress
+        this->CalculateMaterialResponse(data);
+        // Add limits to the strain, and recompute stress if required
+        const auto &constitutive_law = this->GetConstitutiveLaw();
+        auto strain = data.StrainRate;
+        const double gamma_dot = std::sqrt(2.0 * (strain[0] * strain[0] + strain[1] * strain[1] + strain[2] * strain[2]) + strain[3] * strain[3] + strain[4] * strain[4] + strain[5] * strain[5]);
+        double gamma_dot_from_constitutive_law = constitutive_law->GetValue(EQ_STRAIN_RATE, gamma_dot_from_constitutive_law);
+        double ratio = fabs(gamma_dot_from_constitutive_law / gamma_dot);
+        if (ratio < 0.99999 && ratio > 1e-20) {

```

Comment:
In the situations where constitutive law uses a different measure (formula) for eq. strain rate.  Also in case, a constitutive law doesn't compute eq. strain rate. (far-fetched, but possible!).

Reply:
If you want to consider this case calculation must be always done through the constitutive law.

Reply:
Changes done.